lib/diagnostic.js

```diff
@@ -4,18 +4,33 @@
 // Write a callback for reduce that selects
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
-// `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
+// `let minimum = array.reduce(min);
 
+let previousArr = [1, 2, 3, 4, 5];
+let smallestNum = previousArr[0];
+
+const min = function (previousArr, smallestNum) {
```

Comment:
This function is being passed into an invocation of ```reduce()``` as a callback function.  The three parameters for ```reduce()```'s callback are the accumulator (or previous element, as it is in this case), the current element, and the current working array (which is omitted in this case).  For this particular example, ```previous``` and ```current``` are assumed to be numbers.  Here you are assuming that the first parameter is an array when it should be a number, so trying to get the length property of the first argument in your ```for``` loop will throw an error.